### PR TITLE
Mitaka nfp 13 may 2016 vpn testcases

### PR DESCRIPTION
**Pull request by sdharav**
@njagadish  please review the changes.

### CODE REVIEW

File: gbpservice/neutron/tests/unit/nfp/configurator/drivers/vpn/vyos/test_vpn_driver.py [hunk 1]
```diff
@@ -0,0 +1,445 @@
...
+
+            mock_post.assert_called_with(
+                self.dict_objects.url_create_ipsec_tunnel,
+                data=jsonutils.dumps(self.dict_objects.data_),
+                timeout=self.dict_objects.timeout)
+            mock_update_status.assert_called_with(
+                context,
+                self.dict_objects.ipsec_vpnsvc_status)
+
+    def delete_ipsec_site_conn(self):
```
Comment (njagadish) [inline gbpservice/neutron/tests/unit/nfp/configurator/drivers/vpn/vyos/test_vpn_driver.py]: This unit test name needs to be corrected and tested.

File: gbpservice/neutron/tests/unit/nfp/configurator/drivers/vpn/vyos/test_vpn_driver.py [hunk 2]
```diff
@@ -0,0 +1,445 @@
...
+        kwargs = self.dict_objects._make_resource_data(operation='delete',
+                                                       service_type='ipsec')
+        with mock.patch.object(self.plugin_rpc, 'update_status'),\
+                mock.patch.object(json, 'loads') as mock_resp,\
+                mock.patch.object(requests, 'delete') as (
+                mock_delete):
+            mock_resp.return_value = self.fake_resp_dict
+            mock_delete.return_value = self.resp
+            self.driver.vpnservice_updated(self.context, kwargs)
+            mock_delete.assert_called_with(
```
Comment (njagadish) [inline gbpservice/neutron/tests/unit/nfp/configurator/drivers/vpn/vyos/test_vpn_driver.py:122]: Update status function needs to be invoked.

File: gbpservice/neutron/tests/unit/nfp/configurator/drivers/vpn/vyos/test_vpn_driver.py [hunk 3]
```diff
@@ -0,0 +1,445 @@
...
+
+        self.resp = mock.Mock(status_code=200)
+        kwargs = self.dict_objects._make_resource_data(operation='delete',
+                                                       service_type='ipsec')
+        with mock.patch.object(self.plugin_rpc, 'update_status'),\
+                mock.patch.object(json, 'loads') as mock_resp,\
+                mock.patch.object(requests, 'delete') as (
+                mock_delete):
+            mock_resp.return_value = self.fake_resp_dict
+            mock_delete.return_value = self.resp
```
Comment (njagadish) [inline gbpservice/neutron/tests/unit/nfp/configurator/drivers/vpn/vyos/test_vpn_driver.py:111]: Isn't there post() method for delete of ipsec connection? Can we mock if it is there?

File: gbpservice/neutron/tests/unit/nfp/configurator/drivers/vpn/vyos/test_vpn_driver.py [hunk 4]
```diff
@@ -0,0 +1,445 @@
...
+        of generic config driver.
+
+        Returns: none
+
+        """
+
+        with mock.patch.object(
+                requests, 'post', return_value=self.resp) as mock_post, \
+            mock.patch.object(
+                requests, 'delete', return_value=self.resp) as mock_delete:
```
Comment (njagadish) [inline gbpservice/neutron/tests/unit/nfp/configurator/drivers/vpn/vyos/test_vpn_driver.py:244]: Check if delete is also a post or a delete

File: gbpservice/neutron/tests/unit/nfp/configurator/drivers/vpn/vyos/test_vpn_driver.py [hunk 5]
```diff
@@ -0,0 +1,445 @@
...
+    def test_get_success(self):
+        '''
+        Implements testcase for vpn drivers get methode to test in
+        fail condition while making call to the service VM
+        '''
+
+        self.resp = mock.Mock(status_code=200)
+        with mock.patch.object(requests, 'get', return_value=self.resp) as (
+                mock_get):
+            self.rest_obj.get('create-ipsec-site-tunnel', self.data)
```
Comment (njagadish) [inline gbpservice/neutron/tests/unit/nfp/configurator/drivers/vpn/vyos/test_vpn_driver.py:434]: Modify the get method name.

File: gbpservice/neutron/tests/unit/nfp/configurator/drivers/vpn/vyos/test_vpn_driver.py [hunk 6]
```diff
@@ -0,0 +1,445 @@
...
+    def test_get_fail(self):
+        '''
+        Implements testcase for vpn drivers get methode to test in
+        fail condition
+        '''
+
+        self.resp = mock.Mock(status_code=404)
+        with mock.patch.object(requests, 'get', return_value=self.resp) as (
+                mock_get):
+            self.rest_obj.get('create-ipsec-site-tunnel', self.data)
```
Comment (njagadish) [inline gbpservice/neutron/tests/unit/nfp/configurator/drivers/vpn/vyos/test_vpn_driver.py:449]: Change the method name to get

File: gbpservice/neutron/tests/unit/nfp/configurator/test_data/vpn_test_data.py [hunk 7]
```diff
@@ -0,0 +1,566 @@
...
+            return self._delete_ipsec_site_conn_obj()
+        if operation is 'update':
+            return self._update_ipsec_site_conn_obj()
+
+        if operation == 'create' and service_type == 'ipsec':
+            return self._create_ipsec_site_conn_obj()
+        else:
+            return self._create_vpnservice_obj()
+
+    def _fake_resource_data(self):
```
Comment (njagadish) [inline gbpservice/neutron/tests/unit/nfp/configurator/test_data/vpn_test_data.py]: Remove "_" from all the functions used from different classes.

File: gbpservice/nfp/config_orchestrator/agent/vpn.py [hunk 8]
```diff
@@ -66,13 +66,9 @@ def _get_vpn_context(self, context, tenant_id, vpnservice_id,
...
                 'ipsec_site_conns': ipsec_site_conns}
 
-    def _get_core_context(self, context, tenant_id):
-        filters = {'tenant_id': [tenant_id]}
-        core_context_dict = common.get_core_context(context,
-                                                    filters,
-                                                    self._conf.host)
-        del core_context_dict['ports']
-        return core_context_dict
+    def _get_core_context(self, context):
```
Comment (AkashDeepSrivastava) [inline gbpservice/nfp/config_orchestrator/agent/vpn.py:69]: Here we are fetching all the routers and networks information of  all the tenants, one better way is to do two rpc one for service_tenant and 2nd for user_tenant  whoever is creating it.

Comment (sdharav) [inline gbpservice/nfp/config_orchestrator/agent/vpn.py:69]: I guess that will cost the double time right ? both the times the api fetches all networks from all tenants and then we are filtering based on the tenant_id, here we are doing only once.

Comment (AkashDeepSrivastava) [inline gbpservice/nfp/config_orchestrator/agent/vpn.py:69]: yes ,suresh you  are correct,we have to change there first .i am not sure there we can change or not ,need to look at that.with present code this is correct

**Pull request closed without merging**